lib/Common/Memory/Recycler.h

```diff
@@ -739,12 +745,12 @@ class Recycler
             _recycler(recycler),
             _exitState(exitState)
         {
-            _recycler->collectionState = entryState;
+            _recycler->SetCollectionState(entryState);
```

Comment:
Strictly speaking, since collectionState has the assignment operator overloaded, you dont need to have a setter function anymore? but it's probably good to just leave that as is

Reply:
Yeah, I had done this before I introduced `ObservableCollectionState`, and then just decided to leave as-is.